textpipe/pipeline.py

```diff
@@ -58,8 +68,25 @@ def __call__(self, raw):
         """
         doc = Doc(raw, language=self.language, hint_language=self.hint_language,
                   spacy_nlps=self._spacy_nlps)
-        result_dict = {oper.__class__.__name__: oper(doc) for oper in self._operations}
-        return result_dict
+
+        data = {}
+
+        for oper, settings in self.steps:
+            target_operation = self._operations[oper]
+            assert callable(target_operation)
+            data[oper] = target_operation(doc, **{'context': data, 'settings': settings})
```

Comment:
why not pass this as arguemnts?
```suggestion
            data[oper] = target_operation(doc, context=data, settings=settings)
```